### ISSUE TITLE
support per-hero configurations

### ISSUE THREAD

**Issue by FlorianWendelborn** [dodekeract/manta-config-engine-app#23]
- [ ] write to student-chat on switch
- [ ] can be bound to a key or using a hero switcher
- [ ] are essentially just another layout, but inherit everything from default layout if activated (checkbox)
- [ ] include custom settings & co

Will be quite similar to @Sembrani / [D2HeroKeys](https://github.com/Sembrani/D2HeroKeys)

**Comment by Sembrani** [comment]
If you want help, let me know. I'm not too familiar with .js, but am happy to help with planning and debugging the .cfg files.

**Comment by FlorianWendelborn** [comment]
I'm actually trying to plan how to structure the `preset.json` file, which can describe all features this would need.

[Here](https://github.com/dodekeract/manta-config-engine/blob/master/presets/default.json) is an example for the kind of file I currently use for this.

I thought about [something like this](http://i.imgur.com/g4C6knj.jpg), because I actually need to provide layouts for:
- general purpose
- per-hero
- macros like the hero-select-ui & [stuff like this](http://www.initialcommit.org/d2vgs/sheet)

These layouts should also be "inheritable" from other ones, so the preset doesn't contain 100 copies of the hero-select-ui.

**Comment by FlorianWendelborn** [comment]
I think I can just implement this as a pre-defined keyboard-layout. Only thing that bugs me is that it actually has just sooo many rules/keymappings. I have reached `d` and already see that I'm in line 95. :unamused:

**Comment by FlorianWendelborn** [comment]
@Sembrani could you tell me which bindings arc-warden needs? I can't find him in your project.

**Comment by Sembrani** [comment]
It should have been updated, otherwise ill have to get it tomorrow, am busy
for the rest of the night (Australia). If you have anything that's really
repetitive to write, even across files, give me a template and I'll write a
bash script for it. (only way I could have done my project)

**Comment by FlorianWendelborn** [comment]
@Sembrani I'm nearly done, so it doesn't matter anymore, but thanks for the offer. I actually found some errors in your code while manually re-writing it.
1. earth = es, you can alias one of these files
2. shadow should backspace to s

**Comment by FlorianWendelborn** [comment]
Didn't actually take that long, when taking into account how much code that is. :smile:
dodekeract/manta-config-engine@bba45bdac9aea64c39f18d9e9878da4b881717b9

Now I only need to make it actually do something. :confused:

**Comment by Sembrani** [comment]
Arc Warden is implemented in the CoreConfigEngine branch which was designed to be copied on top of Loopuleasa's binds before they were integrated. ~~Am moving it over now.~~ Has now been fixed.

Shadow backspacing to s has been fixed, Earth == ES is there so that when future heroes are added, no re-factoring has to happen. Redundant file for now, but hopefully not for long.

Thanks for catching that.

If you do need a repetitive file(s) created, let me know, or try using the .sh script that I have in / D2HeroKeys. I left every script that I used in it, so there are examples to try to wrap your head around.
